cmd/werf/stages/copy/copy.go

```diff
@@ -0,0 +1,272 @@
+
+func NewCmd(ctx context.Context) *cobra.Command {
+	ctx = common.NewContextWithCmdData(ctx, &commonCmdData)
+	cmd := common.SetCommandContext(ctx, &cobra.Command{
+		Use:                   "copy",
+		Short:                 "Copy stages between container registry and archive",
+		Long:                  common.GetLongCommandDescription(GetCopyDocs().Long),
+		DisableFlagsInUseLine: true,
+		Annotations: map[string]string{
+			common.CmdEnvAnno: common.EnvsDescription(),
+			common.DocsLongMD: GetCopyDocs().LongMD,
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+
+			defer global_warnings.PrintGlobalWarnings(ctx)
+
+			if err := common.ProcessLogOptions(&commonCmdData); err != nil {
+				common.PrintHelp(cmd)
+				return err
+			}
+
+			common.LogVersion()
+
+			return common.LogRunningTime(func() error { return runCopy(ctx) })
+		},
+	})
+
+	common.SetupDir(&commonCmdData, cmd)
+	common.SetupGitWorkTree(&commonCmdData, cmd)
+	common.SetupConfigTemplatesDir(&commonCmdData, cmd)
+	common.SetupConfigPath(&commonCmdData, cmd)
+
+	common.SetupEnvironment(&commonCmdData, cmd)
+
+	common.SetupGiterminismConfigPath(&commonCmdData, cmd)
+	common.SetupGiterminismOptions(&commonCmdData, cmd)
+
+	common.SetupTmpDir(&commonCmdData, cmd, common.SetupTmpDirOptions{})
+	common.SetupHomeDir(&commonCmdData, cmd, common.SetupHomeDirOptions{})
+
+	common.SetupRepoOptions(&commonCmdData, cmd, common.RepoDataOptions{})
+	common.SetupSecondaryStagesStorageOptions(&commonCmdData, cmd)
+	common.SetupCacheStagesStorageOptions(&commonCmdData, cmd)
+	common.SetupFinalRepo(&commonCmdData, cmd)
+
+	common.SetupDockerConfig(&commonCmdData, cmd, "Command needs granted permissions to read, pull and push images into the specified repos")
+	common.SetupInsecureRegistry(&commonCmdData, cmd)
+	common.SetupSkipTlsVerifyRegistry(&commonCmdData, cmd)
+	common.SetupContainerRegistryMirror(&commonCmdData, cmd)
+
+	common.SetupSynchronization(&commonCmdData, cmd)
+
+	common.SetupLogOptions(&commonCmdData, cmd)
+	common.SetupLogProjectDir(&commonCmdData, cmd)
+
+	common.SetupParallelOptions(&commonCmdData, cmd, common.DefaultBuildParallelTasksLimit)
+
+	commonCmdData.SetupFinalImagesOnly(cmd, false)
+	commonCmdData.SetupPlatform(cmd)
+
+	setupCopyOptions(cmd)
+
+	return cmd
+}
+
+func runCopy(ctx context.Context) error {
+	commonManager, ctx, err := common.InitCommonComponents(ctx, common.InitCommonComponentsOptions{
+		Cmd:                         &commonCmdData,
+		InitWerf:                    true,
+		InitGitDataManager:          true,
+		InitDockerRegistry:          true,
+		InitProcessContainerBackend: true,
+		InitManifestCache:           true,
+		InitLRUImagesCache:          true,
+		InitTrueGitWithOptions: &common.InitTrueGitOptions{
+			Options: true_git.Options{LiveGitOutput: *commonCmdData.LogDebug},
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("component init error: %w", err)
+	}
+
+	opts, err := getCopyOptions()
```

Comment:
Please, pass [cmdData](https://github.com/werf/werf/pull/7209/files#diff-b6ffb79c53a3630b60db5e425fa941c73af8f6cc8d7fa4c4dc84d6141b05632dR25) as the argument ino getCopyOptions() and into validateRawCopyOptions() to simplify reading and testing of these functions.